src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -1,449 +0,0 @@
-                // delegate.  It will fall out of the GenerateFieldDelegatingConstructor above.
-                for (var i = _state.Arguments.Length; i >= 1; i--)
-                {
-                    var edit = await GenerateThisOrBaseDelegatingConstructorAsync(i, fieldNamingRule, parameterNamingRule).ConfigureAwait(false);
-                    if (edit.document != null)
-                    {
-                        return edit;
-                    }
-                }
-
-                return default;
-            }
-
-            private async Task<(Document document, bool addedFields)> GenerateThisOrBaseDelegatingConstructorAsync(
-                int argumentCount, NamingRule fieldNamingRule, NamingRule parameterNamingRule)
-            {
-                (Document document, bool addedField) edit;
-                if ((edit = await GenerateDelegatingConstructorAsync(argumentCount, _state.TypeToGenerateIn, fieldNamingRule, parameterNamingRule).ConfigureAwait(false)).document != null ||
-                    (edit = await GenerateDelegatingConstructorAsync(argumentCount, _state.TypeToGenerateIn.BaseType, fieldNamingRule, parameterNamingRule).ConfigureAwait(false)).document != null)
-                {
-                    return edit;
-                }
-
-                return default;
-            }
-
-            private async Task<(Document, bool addedFields)> GenerateDelegatingConstructorAsync(
-                int argumentCount, INamedTypeSymbol namedType, NamingRule fieldNamingRule, NamingRule parameterNamingRule)
-            {
-                if (namedType == null)
-                {
-                    return default;
-                }
-
-                // We can't resolve overloads across language.
-                if (_document.Project.Language != namedType.Language)
-                {
-                    return default;
-                }
-
-                var arguments = _state.Arguments.Take(argumentCount).ToList();
-                var remainingArguments = _state.Arguments.Skip(argumentCount).ToImmutableArray();
-                var remainingAttributeArguments = _state.AttributeArguments != null
-                    ? _state.AttributeArguments.Skip(argumentCount).ToImmutableArray()
-                    : (ImmutableArray<TAttributeArgumentSyntax>?)null;
-                var remainingParameterTypes = _state.ParameterTypes.Skip(argumentCount).ToImmutableArray();
-
-                var instanceConstructors = namedType.InstanceConstructors.Where(IsSymbolAccessibleToDocument).ToSet();
-                if (instanceConstructors.IsEmpty())
-                {
-                    return default;
-                }
-
-                var delegatedConstructor = _service.GetDelegatingConstructor(_state, _document, argumentCount, namedType, instanceConstructors, _cancellationToken);
-                if (delegatedConstructor == null)
-                {
-                    return default;
-                }
```

Comment:
the logic above here moved into determining which code actions we want.  the logic below moved into the code action itself to apply the change.